src/commands/Minion/cancel.ts

```diff
@@ -0,0 +1,68 @@
+import { KlasaMessage, CommandStore } from 'klasa';
+
+import { BotCommand } from '../../lib/BotCommand';
+import getActivityOfUser from '../../lib/util/getActivityOfUser';
+import removeSubTasksFromActivityTask from '../../lib/util/removeSubTasksFromActivityTask';
+import { Tasks, Activity, Time } from '../../lib/constants';
+import { requiresMinion } from '../../lib/minions/decorators';
+
+const options = {
+	max: 1,
+	time: 10000,
+	errors: ['time']
+};
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			oneAtTime: true,
+			cooldown: 180
+		});
+	}
+
+	@requiresMinion
+	async run(msg: KlasaMessage) {
+		if (!msg.author.minionIsBusy) {
+			return msg.send(`${msg.author.minionStatus}`);
+		}
+
+		const currentTask = getActivityOfUser(this.client, msg.author);
+
+		if (!currentTask) {
+			throw `Your minion has no current task`;
+		}
+
+		if (currentTask.finishDate - Date.now() < Time.Minute * 2) {
+			throw `You can not cancel a task with less than two minutes remaining`;
+		}
+
+		if (currentTask.type === Activity.GroupMonsterKilling) {
+			throw `You can not leave a group activity`;
+		}
+
+		const taskTicker =
+			currentTask.type === Activity.ClueCompletion
+				? Tasks.ClueTicker
+				: currentTask.type === Activity.MonsterKilling
+				? Tasks.MonsterKillingTicker
+				: Tasks.SkillingTicker;
+
+		const cancelMsg = await msg.channel.send(
+			`${msg.author} ${msg.author.minionStatus}\n Say \`confirm\` to cancel this activity. you will recieve **NO** reimbursment of items consumed when your minion was sent out on this activity.`
```

Comment:
`${msg.author} ${msg.author.minionStatus}\n Say \`confirm\` if you want to call your minion back from their trip. They'll drop all their current loot to get back as fast as they can, so you won't receive any loot from this trip if you cancel it.`